storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestDefaultResourceAwareStrategy.java

```diff
@@ -905,9 +911,9 @@ public void testMultipleRacksWithFavoritism() {
 
         rs.prepareForScheduling(cluster, topo1);
         INodeSorter nodeSorter = new NodeSorter(cluster, topo1, BaseResourceAwareStrategy.NodeSortType.DEFAULT_RAS);
-        TreeSet<ObjectResourcesItem> sortedRacks= nodeSorter.sortRacks(null);
+        nodeSorter.prepare(null);
+        Iterable<ObjectResourcesItem> sortedRacks= nodeSorter.getSortedRacks();
 
-        Assert.assertEquals("# of racks sorted", 5, sortedRacks.size());
```

Comment:
Why do we want to remove this check?

Reply:
The returned object it iterable. The check happens implicitly when each of the five elements are checked following.